prql-php/src/Compiler.php

```diff
@@ -39,34 +39,27 @@ final class Compiler
      */
     function __construct(?string $lib_path = null)
     {
-        $library = $lib_path;
-
         if ($lib_path === null) {
-            $library = __DIR__;
+            $lib_path = __DIR__ . "/../lib";
         }
 
+        $header = $lib_path . "/libprql_lib.h";
+
         if (PHP_OS_FAMILY === "Windows") {
-            $library .= "\libprql_lib.dll";
+            $library = $lib_path . "\libprql_lib.dll";
         } elseif (PHP_OS_FAMILY === "Darwin") {
-            $library .= "/libprql_lib.dylib";
+            $library = $lib_path . "/libprql_lib.dylib";
         } else {
-            $library .= "/libprql_lib.so";
+            $library = $lib_path . "/libprql_lib.so";
+        }
+
+        $header_source = file_get_contents($header, false, null, 0, 1024 * 1024);
+
+        if ($header_source === false) {
+            throw new \InvalidArgumentException("Cannot load header file.");
         }
 
-        $this->_libprql = \FFI::cdef(
-            "
-            typedef struct Options {
-                bool format;
-                char *target;
-                bool signature_comment;
-            } Options;
-
-            int compile(const char *prql_query, const struct Options *options, char *out);
-            int prql_to_pl(const char *prql_query, char *out);
-            int pl_to_rq(const char *pl_json, char *out);
-            int rq_to_sql(const char *rq_json, char *out);
-        ", $library
-        );
+        $this->_libprql = \FFI::cdef($header_source, $library);
```

Comment:
You can use the [`FFI:load`](https://www.php.net/manual/en/ffi.load.php) to have it load the file for you, instead of load it with `file_get_contents` and pass it a string.

Reply:
I looked at FFI::load, but it does not load the library itself, only the header. You have to define FFI_LIB variable to do that, but we are using an auto generated file.